src/naive_bayes/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::error::Failed;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+pub trait NBDistribution {
+    fn fit<T: RealNumber, M: Matrix<T>>(&self, x: &M, y: M::RowVector) -> Self;
+
+    fn prior<T: RealNumber>(k: T) -> T;
+
+    fn conditional_probability<T: RealNumber, M: Matrix<T>>(k: T, j: M::RowVector) -> T;
+}
+
+pub struct BaseNaiveBayes;
+
+impl BaseNaiveBayes {
+    pub fn fit<D: NBDistribution>(&mut self, distribution: D) -> Result<(), Failed> {
+        unimplemented!()
+    }
+
+    pub fn predict<T: RealNumber, M: Matrix<T>>(x: &M) -> Result<M::RowVector, Failed> {
```

Comment:
you might have forgot `self` here
```suggestion
    pub fn predict<T: RealNumber, M: Matrix<T>>(&self, x: &M) -> Result<M::RowVector, Failed> {
```